config/config.go

```diff
@@ -58,8 +67,144 @@ func (c *APIServerConfig) ReadLogFromCloudStorage() bool {
 
 // AuthConfig determines whether to use IAP for authentication and authorization
 type AuthConfig struct {
-	IAP    *IAPAuthConfig `yaml:"iap"`
-	APIKey string         `yaml:"apiKey"`
+	IAP            *IAPAuthConfig   `yaml:"iap"`
+	APIKey         string           `yaml:"apiKey"`
+	OAuthProviders []*OAuthProvider `yaml:"oauthProviders"`
+	JWT            *JWTConfig       `yaml:"jwt"`
+}
+
+// OAuthProvider is used to configure one or more oauth providers like google, github, microsoft
+type OAuthProvider struct {
+	Name                   string `yaml:"name"`
+	ClientID               string `yaml:"clientID"`
+	ClientSecret           string `yaml:"clientSecret"`
+	AllowedIdentitiesRegex string `yaml:"allowedIdentitiesRegex"`
+}
+
+// OAuthProviderInfo provides non configurable information for oauth providers
+type OAuthProviderInfo struct {
+	AuthURL  string
+	TokenURL string
+}
+
+// GetConfig returns the oauth config for the provider
+func (p *OAuthProvider) GetConfig(baseURL string) *oauth2.Config {
+
+	redirectPath := "/api/auth/handle/"
+	redirectURI := strings.TrimSuffix(baseURL, "/") + redirectPath + p.Name
+
+	oauthConfig := oauth2.Config{
+		ClientID:     p.ClientID,
+		ClientSecret: p.ClientSecret,
+		RedirectURL:  redirectURI,
+	}
+
+	switch p.Name {
+	case "google":
+		oauthConfig.Endpoint = endpoints.Google
+		oauthConfig.Scopes = []string{
+			"https://www.googleapis.com/auth/userinfo.email",
+		}
+	case "microsoft":
+		oauthConfig.Endpoint = endpoints.Microsoft
+		oauthConfig.Scopes = []string{
+			"user.read+openid+profile+email",
+		}
+	case "facebook":
+		oauthConfig.Endpoint = endpoints.Facebook
+	case "github":
+		oauthConfig.Endpoint = endpoints.Facebook
+	case "bitbucket":
+		oauthConfig.Endpoint = endpoints.Facebook
+	case "gitlab":
+		oauthConfig.Endpoint = endpoints.Facebook
```

Comment:
not sure if this was intended, but hey are all set to Facebook

Reply:
Error, but they haven't been implemented correctly yet anyway; I need to figure out for all of them where to get the user profile and implement that and map the response to our `UserIdentity` object.